floss/main.py

```diff
@@ -850,70 +852,62 @@ def create_r2_script(sample_file_path, r2_script_file, decoded_strings, stack_st
     # TODO return, catch exception in main()
 
 
-def create_json_output(json_file_path, sample_file_path, decoded_strings, stack_strings):
+def create_json_output(options, sample_file_path, decoded_strings, stack_strings, static_strings):
     """
     Create a report of the analysis performed by FLOSS
-    :param json_file_path: path to write the report
+    :param options: parsed options
     :param sample_file_path: path of the sample analyzed
     :param decoded_strings: list of decoded strings ([DecodedString])
     :param stack_strings: list of stack strings ([StackString])
     """
-    results = {'stack_strings': [sanitize_string_for_printing(ss.s) for ss in stack_strings],
-               'decoded_strings': [sanitize_string_for_printing(ds.s) for ds in decoded_strings]}
-    report = {'file_path': sample_file_path, 'results': results}
+    strings = {'stack_strings': stack_strings,
```

Comment:
Have you explored if strings need to be sanitized here?

Reply:
I gave it some thought and played around with turning the sanitize functions into generators (again to try to keep memory footprint low) but didn't end up finishing that because I couldn't come up with a good way to accomplish it.

I'll spend some more time thinking about how to solve that problem.

Reply:
Resolved by 2521339